tests/lib/c_lib/src/main.c

```diff
@@ -624,10 +624,14 @@ ZTEST(test_c_lib, test_str_operate)
 
 	zassert_true(strncat(ncat, str1, 2), "strncat failed");
 	zassert_not_null(strncat(str1, str3, 2), "strncat failed");
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wstringop-overflow"
+#endif
```

Comment:
`Wstringop-overflow` was added in GCC 7, so this should be 7.

https://github.com/gcc-mirror/gcc/commit/ee92e7bafb52d267f52e4c48edada5f279bcf591

Reply:
Good catch, thank you.